Allow to specify a custom action instead of the star

raimund-schluessler opened the pull request:
This PR allows to specify a custom action instead of the star icon in the sidebar. Useful e.g. for the Tasks app to toggle the completed state of a task.

Since #1288 introduced quite some changes to the structure, I implemented it on top of #1288. The important commit is f1ea31adc8fa131a96d4bdd9609bc9848c70edfe. I will rebase this PR here once #1288 is in.

![custom-action](https://user-images.githubusercontent.com/2496460/90564528-cf2e6880-e1a5-11ea-9856-10ad843eafbe.gif)

raimund-schluessler commented:
#1288 needs to be merged first.

raimund-schluessler commented:
Good for review now.

skjnldsv commented on src/components/AppSidebar/AppSidebar.vue:
```diff
@@ -163,14 +163,17 @@
-							@click.prevent="toggleStarred" />
+					<div v-if="canStar || $slots['tertiary-actions']" class="app-sidebar-header__tertiary-actions">
+						<slot name="tertiary-actions">
+							<a v-if="canStar"
+								:class="{
+									'icon-starred': isStarred && !starLoading,
+									'icon-star': !isStarred && !starLoading,
+									'icon-loading-small': starLoading
+								}"
+								class="app-sidebar-header__star"
+								@click.prevent="toggleStarred" />
```
Nice!

skjnldsv commented:
Shall we add an example for the docs?

raimund-schluessler commented:
> 
> 
> Shall we add an example for the docs?

Fine with me. I will add one.

raimund-schluessler commented:
> Shall we add an example for the docs?

I added an avatar as tertiary action example. Probably doesn't really makes sense to have an avatar there (unless the sidebar is about a user), but it shows the use of the tertiary action.

![Screenshot_2020-08-20 Nextcloud Vue Style Guide](https://user-images.githubusercontent.com/2496460/90733794-56c2c700-e2cd-11ea-8666-3775660132e9.png)

skjnldsv commented:
> I added an avatar as tertiary action example. Probably doesn't really makes sense to have an avatar there (unless the sidebar is about a user), but it shows the use of the tertiary action.

Maybe just add a checkbox?

raimund-schluessler commented:
> 
> 
> > I added an avatar as tertiary action example. Probably doesn't really makes sense to have an avatar there (unless the sidebar is about a user), but it shows the use of the tertiary action.
> 
> Maybe just add a checkbox?

Yes, might be more reasonable. I changed it.

![Screenshot_2020-08-20 Nextcloud Vue Style Guide(1)](https://user-images.githubusercontent.com/2496460/90742329-b621d680-e2cf-11ea-85f8-2be8eea5d27d.png)
